pontoon/base/models.py

```diff
@@ -276,36 +276,39 @@ def repository_for_path(self, path):
         else:
             return repo
 
-    def latest_activity(self, locale=None):
+    def latest_activity(self, locale=None, path=None):
         """Get latest activity for project and locale if provided."""
         translations = Translation.objects.filter(entity__resource__project=self)
         if locale:
             translations = translations.filter(locale=locale)
+            if path:
+                translations = translations.filter(entity__resource__path=path)
 
         return translations.latest_activity()
 
-    @property
-    def locales_parts_stats(self):
+    def locales_parts_stats(self, loc=None):
```

Comment:
It's awkward to have this method living on the project model, especially with a parameter that can limit it to a single locale. It's starting to look more like a method for the locale queryset that takes in a project to filter the stats by. That way you can better control which locales you want to get stats from.